src/libraries/AddressStringUtil.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+/// @title AddressStringUtil
+/// @notice provides utility functions for converting addresses to strings
+library AddressStringUtil {
+    /// @notice Converts an address to the uppercase hex string, extracting only len bytes (up to 20, multiple of 2)
+    /// @param addr the address to convert
+    /// @param len the number of bytes to extract
+    /// @return the hex string
+    function toAsciiString(address addr, uint256 len) internal pure returns (string memory) {
```

Comment:
Is this copied from v3? The ones that have been it would be great to have a comment at the top of the file thats like /// Based on *link to v3 file*

Reply:
bc currently i dont know if i need to review each of these files in fulllll.. like AddressStringUtil, SafeERC20Namer, SVG, HexStrings etc